include/mbedtls/md.h

```diff
@@ -132,220 +138,255 @@ void mbedtls_md_free( mbedtls_md_context_t *ctx );
- *                  Makes it necessary to call mbedtls_md_free() later.
+ * \brief           This function selects the MD to use, and allocates internal 
+ *                  structures.
+ *                  It should be called after mbedtls_md_init() or 
+ *                  mbedtls_md_free(). Makes it necessary to call 
+ *                  mbedtls_md_free() later.
  *
- * \param ctx       Context to set up.
- * \param md_info   Message digest to use.
- * \param hmac      0 to save some memory if HMAC will not be used,
- *                  non-zero is HMAC is going to be used with this context.
+ * \param ctx       The context to set up.
+ * \param md_info   The message digest to use.
+ * \param hmac      <ul><li>0: HMAC is not used. Saves some memory.</li>
+ *                  <li>non-zero: HMAC is used with this context.</li></ul>
  *
  * \returns         \c 0 on success,
- *                  \c MBEDTLS_ERR_MD_BAD_INPUT_DATA on parameter failure,
- *                  \c MBEDTLS_ERR_MD_ALLOC_FAILED memory allocation failure.
+ *                  #MBEDTLS_ERR_MD_BAD_INPUT_DATA on parameter failure, or
+ *                  #MBEDTLS_ERR_MD_ALLOC_FAILED on memory allocation failure.
  */
 int mbedtls_md_setup( mbedtls_md_context_t *ctx, const mbedtls_md_info_t *md_info, int hmac );
 
 /**
- * \brief           Clone the state of an MD context
+ * \brief           This function clones the state of an MD context.
  *
- * \note            The two contexts must have been setup to the same type
- *                  (cloning from SHA-256 to SHA-512 make no sense).
+ * \note            The two contexts must have the same type,
+ *                  for example, both are SHA-256.
  *
- * \warning         Only clones the MD state, not the HMAC state! (for now)
+ * \warning         Only clones the MD state, not the HMAC state.
  *
- * \param dst       The destination context
- * \param src       The context to be cloned
+ * \param dst       The destination context.
+ * \param src       The context to be cloned.
  *
  * \return          \c 0 on success,
- *                  \c MBEDTLS_ERR_MD_BAD_INPUT_DATA on parameter failure.
+ *                  #MBEDTLS_ERR_MD_BAD_INPUT_DATA on parameter failure.
  */
 int mbedtls_md_clone( mbedtls_md_context_t *dst,
                       const mbedtls_md_context_t *src );
 
 /**
- * \brief           Returns the size of the message digest output.
+ * \brief           This function returns the size of the message-digest output.
  *
- * \param md_info   message digest info
+ * \param md_info   The message-digest information.
  *
- * \return          size of the message digest output in bytes.
+ * \return          The size of the message-digest output in Bytes.
  */
 unsigned char mbedtls_md_get_size( const mbedtls_md_info_t *md_info );
 
 /**
- * \brief           Returns the type of the message digest output.
+ * \brief           This function returns the type of the message-digest output.
```

Comment:
Blocker: This is a confusing pre-existing wording - the type of the MD output is just a byte-string. Suggestion: `This function returns the type of the message-digest`, or, more detailed, `This function extracts the message-digest type from the message-digest information structure.` It's inverse to `mbedtls_md_from_type`, which finds the message-digest information structure from the numeric message-digest type.

Reply:
I see that the function `mbedtls_md_get_name` uses the same wording, so we should have some agreement across the team here before changing it everywhere. @RosZad @gilles-peskine-arm @patater?

Reply:
I'm ok with your suggested phrasing: "This function extracts the message-digest type from the message-digest information structure.". let's wait to see what @gilles-peskine-arm and @Patater think, and I'll make the change.

Reply:
@Patater Could you give your opinion that?

Reply:
I'm happy with either of the two suggestions from @hanno-arm.

Reply:
Changed to "This function extracts the message-digest type from the message-digest information structure"